src/games/ecards.py

```diff
@@ -0,0 +1,275 @@
+        - displaying round results to channel
+
+        :param leader: the index of the leader player
+        :param prompt: the prompt for the round
+        :param hands: the players' hands
+        """
+        # announce new round
+        await self.channel.send(
+            embed=self.embed(
+                f"Starting new round -- {self.players[leader]} is leader.\nThis round's prompt: {prompt}"
+            )
+        )
+
+        # immutable pids
+        pids = list(self.players.keys())
+
+        # construct content
+        content_dict = {
+            pid: self.embed(
+                f"This round's prompt: {prompt}\nYou're the leader for this round - sit back and relax!"
+            )
+            if pid == leader
+            else self.embed(f"This round's prompt: {prompt}")
+            for pid in pids
+        }
+
+        # get all message responses
+        tasks = []
+        for pid in pids:
+            # get channel
+            dm_channel = await self.players[pid].create_dm()
+
+            if pid != leader:
+                # make pipeline
+                ipl = InteractionPipeline(ChoiceInteraction(*hands[pid], max_votes=1))
+
+                # send message by storing coroutine
+                tasks.append(ipl.send_and_watch(dm_channel, content_dict[pid]))
+            else:
+                # leader
+                tasks.append(dm_channel.send(embed=content_dict[pid]))
+
+        # await routines
+        dm_response = {
+            # ensure type consistent output
+            pid: resp if pid != leader else {}
+            for pid, resp in zip(pids, await asyncio.gather(*tasks))
+        }
+
+        # unpack which card played
+        # pid -> str
+        cards_played = {} 
+        for pid, resp in dm_response.items():
+
+            # make sure result is present
+            if "choice" in resp.get("response", {}):
+                # invert choices:
+                inverted = {v: k for k, v in resp["response"]["choice"].items()}
+
+                # get, or falsey (min enumeration is 1)
+                index = inverted.get(1, 0)
+
+                if index:
+                    index = inverted[1] - 1
+                    cards_played[pid] = hands[pid].pop(index)
+                    self.logging.info(f"player {pid} chose {index}")
+
+                else:
+                    # no card: log for now
+                    self.logging.info(
+                        f"Player {self.players[pid]} did not play a card."
+                    )
+
+            elif pid != leader:
+                # log warning
+                self.logging.warning(f"Bad response from {self.players[pid]}: {resp}")
+
+        # shuffle responses to list
+        shuffled_responses = list(cards_played.values())
+        random.shuffle(shuffled_responses)
+
+        asyncio.sleep(self.wait_duration)
+
+        if len(shuffled_responses) == 0:
+            # No-one played a card - skip the round
+            await self.channel.send(
```

Comment:
Can you add a `return await self.channel.send(...)` here, so it's obvious the function exits and no more code gets executed further down the page?

Or alternatively
```py
await self.channel.send(...)
return
```